### PR TITLE
Various improvements for the AnnePro2

### PR DESCRIPTION
**Pull request by Jpe230**
Various improvements for the AnnePro2

## Description

This PR implements these improvements:
- Add a RGB driver that utilizes the current communication protocol between the main MCU and the LED MCU (i.e. QMK animation)
- Renamed various variables and functions name to be snake-case compliant
- Improve readability in the default keymaps
- Enable community layout support

My PR currently only aims to update 1 of the 3 default keymaps to enable the RGB driver, the other 2 stay with the current way of handling the RGB as they serve the purpose of showcasing in how to use the protocol.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/annepro2/config_led.c [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#include "rgb_matrix.h"
```
Comment (drashna) [inline keyboards/annepro2/config_led.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/annepro2/config_led.h [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#pragma once
```
Comment (drashna) [inline keyboards/annepro2/config_led.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/annepro2/rgb_driver.c [hunk 3]
```diff
@@ -0,0 +1,46 @@
+#include "rgb_matrix.h"
```
Comment (drashna) [inline keyboards/annepro2/rgb_driver.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/annepro2/annepro2.c [hunk 4]
```diff
@@ -99,6 +101,8 @@ void keyboard_post_init_kb(void) {
 
     annepro2LedGetStatus();
 
+    annepro2LedEnable();
```
Comment (drashna) [inline keyboards/annepro2/annepro2.c:104]: Not a fan of the casing, but that's mostly a nitpick.

Comment (Jpe230) [reply keyboards/annepro2/annepro2.c:104]: I agree,  maybe a good PR would be to standardize the casing in this keyboard.

**Review by tzarc — APPROVED** [review]

**Pull request merged**